core/src/main/java/hudson/model/Executor.java

```diff
@@ -220,6 +220,13 @@ public void run() {
             try {
                 workUnit.context.synchronizeStart();
 
+                // this code handles the behavior of null Executables returned
+                // by tasks. In such case Jenkins starts the workUnit in order 
+                // to report results to console outputs. 
+                if (executable == null) {
+                    throw new Error("The null Executable has been created for "+workUnit+". The task cannot be executed");
```

Comment:
Correction: it does say what the task was, but not in a very readable way. At any rate, I am improving in #2963.